functions/Find-DbaOrphanedFile.ps1

```diff
@@ -0,0 +1,231 @@
+﻿
+Function Find-DbaOrphanedFile
+{
+<#
+.SYNOPSIS 
+Find-DbaOrphanedFile moves/removes orphaned database files
+
+.DESCRIPTION
+Get all the database files for all the database for the instance
+Get the various directories of the instance and get all the present database files.
+Compare which the two lists to see if there are any orphaned files and return the list
+
+.PARAMETER SqlServer
+The SQL Server instance. You must have sysadmin access and server version must be SQL Server version 2000 or higher.
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. 
+
+.PARAMETER SourceDirectory
+Used to specify extra directories to search through besides the default data and log directories
+
+.NOTES 
+Author: Sander Stad (@sqlstad), sqlstad.nl
+Requires: sysadmin access on SQL Servers
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+
+Copyright (C) 2016 Chrissy LeMaire
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Find-DbaOrphanedFile
+
+.EXAMPLE
+Find-DbaOrphanedFile -SqlServer sqlserver2014a
+Copies all policies and conditions from sqlserver2014a to sqlcluster, using Windows credentials. 
+
+.EXAMPLE   
+Find-DbaOrphanedFile -SqlServer sqlserver2014a -SqlCredential $cred
+Does this, using SQL credentials for sqlserver2014a and Windows credentials for sql instance.
+
+.EXAMPLE   
+Find-DbaOrphanedFile -SqlServer sqlserver2014 -Sourcedirectory 'C:\Dir1', 'C:\Dir2'
+Finds the orphaned files in the default directories but also the extra ones
+	
+
+#>
+	
+	# This is a sample. Please continue to use aliases for discoverability. Also keep the [object] type for sqlserver.
+	[CmdletBinding(SupportsShouldProcess = $true)]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[object]$SqlServer,
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[object]$SqlCredential,
+        [parameter(Mandatory = $false, ValueFromPipeline = $true)]
+        [string[]]$SourceDirectory

```

Comment:
I will rename the SourceDirectory param to Path, no problem.
Yes, this path will be in addition to the defaults in some cases we have databases in other directories or drives that are not the default in SQL Server.